### PR TITLE
add --token: Bearer token for authentication to the Kubernetes API  #4026

### PR DESCRIPTION
**Pull request by humboldt-xie**
add --token: Bearer token for authentication to the Kubernetes API server

    https://github.com/linkerd/linkerd2/issues/4026

    Signed-off-by: humboldt <humboldt_xie@163.com>

### CODE REVIEW

File: pkg/k8s/api.go [hunk 1]
```diff
@@ -82,6 +92,20 @@ func NewAPI(configPath, kubeContext string, impersonate string, impersonateGroup
 		TsClient:      tsClient,
 		DynamicClient: dynamicClient,
 	}, nil
+
+}
+
+// NewAPI validates a Kubernetes config and returns a client for accessing the
+// configured cluster.
+func NewAPI(configPath, kubeContext string, impersonate string, impersonateGroup []string, timeout time.Duration) (*KubernetesAPI, error) {
```
Comment (cpretzer) [inline pkg/k8s/api.go]: Is this function used anywhere? I see only calls to `newK8sAPI` and subsequent calls to `NewAPIForConfig` in the modified files where `NewAPI` was previously called.

Comment (humboldt-xie) [reply pkg/k8s/api.go]: Keep NewAPI for compatibility, add NewAPIConfig for more extensibility

**Comment by adleong** [comment]
Hi @humboldt-xie, thanks for this PR!  I left a question of the associated issue: https://github.com/linkerd/linkerd2/issues/4026#issuecomment-584867322 please take a look.

**Comment by cpretzer** [comment]
hi @humboldt-xie I'd like to get this PR merged, if you're still interested in working on the comments that @adleong left.

**Comment by cpretzer** [comment]
@humboldt-xie I ran some tests locally with this PR and the automated tests all seem to run fine

I'm curious about the expected behavior when the `--token` flag is provided. Specifically, when I send an invalid value, the commands still seem to work as expected.

For example #4026 includes the example:
`linkerd check --token "balabala"`

When I run that locally after installing Linkerd, the command is executed as expected. So, it's not clear to me the point at which the token is validated.

**Comment by grampelberg** [comment]
@humboldt-xie are you still working on this or should we close it out?

**Pull request closed without merging**